module/documents/active-effect.mjs

```diff
@@ -142,6 +143,23 @@ export default class ActiveEffect5e extends ActiveEffect {
       return current;
     }
 
+    // If attempting to apply active effect to empty MappingField entry, create it
+    if ( (current === undefined) && change.key.startsWith("system.") ) {
+      let mappingField = field;
+      const [, ...parts] = change.key.split(".");
+      let last;
+      while ( !(mappingField instanceof MappingField) ) {
+        last = parts.pop();
+        mappingField = model.system.schema.getField(parts.join("."));
+        if ( !mappingField || !parts.length ) break;
+      }
+      const keyPath = ["system", ...parts, last].join(".");
+      if ( mappingField && (foundry.utils.getProperty(model, keyPath) === undefined) ) {
+        const created = mappingField.model.initialize(mappingField.model.getInitialValue(), mappingField);
+        foundry.utils.setProperty(model, keyPath, created);
+      }
```

Comment:
Any disadvantage to using the `parent` chain here instead? I think it's probably faster, which is important if it happens during preparation.

```suggestion
      let mappingField = field;
      while ( !(mappingField instanceof MappingField) && mappingField ) mappingField = mappingField.parent;
      if ( mappingField && (foundry.utils.getProperty(model, mappingField.fieldPath) === undefined) ) {
        const created = mappingField.model.initialize(mappingField.model.getInitialValue(), mappingField);
        foundry.utils.setProperty(model, mappingField.fieldPath, created);
      }
```

Reply:
MappingField wasn't properly setting itself as the `parent` of its containing model.